### PR TITLE
Support for server-sent events / streaming

### PR DESCRIPTION
**Pull request by jkodumal**

### CODE REVIEW

File: lib/ldclient-rb/config.rb [hunk 1]
```diff
@@ -30,6 +31,9 @@ def initialize(opts = {})
       @connect_timeout = opts[:connect_timeout] || Config.default_connect_timeout
       @read_timeout = opts[:read_timeout] || Config.default_read_timeout
       @log_timings = opts[:log_timings] || Config.default_log_timings
+      @stream = opts[:stream] || Config.default_stream
+      @feature_store = opts[:feature_store] || Config.default_feature_store
+      @debug_stream = opts[:debug_stream] || false
```
Comment (pkaeding) [inline lib/ldclient-rb/config.rb]: Any reason not to have a `Config.default_debug_stream` here?

**Comment by apucacao** [comment]
lgtm

File: lib/ldclient-rb/stream.rb [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    def set_connected()
+      @disconnected.set(nil)
+    end
+
+    def should_fallback_update()
+      disc = @disconnected.get
+      disc != nil and disc < (Time.now - 120)
+    end
+
+    # TODO mark private methods
```
Comment (pkaeding) [inline lib/ldclient-rb/stream.rb]: I think this `TODO` can be removed

**Comment by pkaeding** [comment]
lgtm

**Pull request merged**